custom_components/coway/sensor.py

```diff
@@ -320,8 +324,8 @@ def has_entity_name(self) -> bool:
     @property
     def native_value(self) -> int:
         """Return current PM10 measurement."""
-
-        return self.purifier_data.particulate_matter_10
+        pm10 = self.purifier_data.particulate_matter_10
+        return pm10 if pm10.isnumeric() else 0
```

Comment:
Since the PM10 sensor isn't available for the IconS, I'll handle removing the sensor from being added for any purifiers that are the IconS - the sensor always returns a value for 300s and 400s models so doing this check is unnecessary. With the change of the sensor not being added for IconS purifiers, you won't have any issues with Home Assistant complaining about non-numeric numbers.